modules/steam/updater.nix

```diff
@@ -0,0 +1,64 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    mkIf
+    mkMerge
+    mkOption
+    optionalString
+    types
+  ;
+
+  cfg = config.jovian.steam;
+in
+{
+  options = {
+    jovian.steam.updater = {
+      splash = mkOption {
+        type = types.enum [
+          "steamos"
+          "jovian"
+          "vendor"
+        ];
+        default = "steamos";
+        description = ''
+          Configures the source of the splash screen used by the updater (preloader) step when launching Steam.
+
+          When `steamos`, this will use the vendor-selected image, scaled appropriately.
+
+          When `jovian`, this will use the Jovian Experiments logo, scaled appropriately.
+
+          When `vendor`, the vendor default will not be changed. This differs from `default` in that
+          on systems other than the Steam Deck, the scaling may not be correct.
+
+          > The scale of the vendor logo (in `vendor`) is incorrect when the display resolution
+          > is different from 1280 pixels wide. The updater logo will be bigger than the one
+          > used in the startup animation.
+        '';
+      };
+    };
+  };
+  config = mkIf (cfg.enable && cfg.updater.splash != "vendor") (mkMerge [
+    {
+      environment.etc."xdg/gamescope-session/environment" = {
```

Comment:
Maybe we should stick most of this in a systemd service instead? And have gamescope just look in /run/jovian

Reply:
I agree, though I still believe hooking into the `environment` affordance like this *to configure the logo* is the proper thing to do. So the logo would be generated by a service instead of within that configuration file.

It would be more inconvenient to patch out the `ui_background` variable from the vendor script in a robust manner, wouldn't it?

Reply:
Yeah, the envvars part is fine, I just don't want to run imagemagick on every Steam startup.